File: leetcode2/1easy/이준열/1437.cpp
class Solution {
    public:
        bool kLengthApart(vector<int>& nums, int k) {
            int zeroCount = 0;
            int oneIndex = -1;
            for (int i = 0;  i < nums.size(); i++)
            {
                if (nums[i] == 1)
                {
                    if (oneIndex == -1)
                    {
                        oneIndex = 0;
                    }
                    else
                    {
                        if (zeroCount < k)
                            return false;
                    }
                    zeroCount = 0;
                }
                else if (nums[i] == 0)
                {
                    zeroCount++;
                }
            }
            return true;
        }
    };

File: leetcode2/1easy/유시연/1437.py
from typing import List

class Leetcode.이재훈.이재훈.Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        dis = 0
        found_first_one = False  

        for i in range(n):
            if nums[i] == 1:
                if not found_first_one:
                    found_first_one = True
                else:
                    if dis < k:
                        return False
                dis = 0 
            else:
                if found_first_one:
                    dis += 1

        return True
